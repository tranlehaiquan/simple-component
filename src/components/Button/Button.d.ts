import * as react from 'react';

export interface ButtonProps {
  autoFocus: boolean,
  disabled: boolean,
  type: string,
  nativeType: 'button'| 'submit'| 'reset',
  children: React.ReactNode,
  onClick: (event: React.MouseEvent) => void
}

declare const Button: React.ComponentType<ButtonProps>;

export default Button;
