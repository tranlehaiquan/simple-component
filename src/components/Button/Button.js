import React, { memo } from 'react';
import propTypes from 'prop-types';
import classnames from 'classnames';

function Button(props) {
  const {
    autoFocus,
    disabled,
    nativeType,
    type,
    children,
    ...restProps
  } = props;

  const btnTypeClassname = type ? `sp-button--${type}` : '';
  const btnClassnames = classnames('sp-button', btnTypeClassname, {
    'sp-button--disabled': disabled,
  });

  return (
    <button
      autoFocus={autoFocus}
      disabled={disabled}
      type={nativeType}
      className={btnClassnames}
      {...restProps}
    >
      {children}
    </button>
  );
}

Button.propTypes = {
  autoFocus: propTypes.bool,
  disabled: propTypes.bool,
  type: propTypes.string,
  nativeType: propTypes.oneOf(['button', 'submit', 'reset']),
  children: propTypes.node,
  // icon: propTypes.string,
  // loading: propTypes.bool
};

Button.defaultProps = {
  autoFocus: false,
  disabled: false,
  type: '',
  nativeType: 'button',
  children: null,
  // icon: '',
  // loading: false
};

export default memo(Button);
