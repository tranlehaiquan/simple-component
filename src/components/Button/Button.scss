.sp_button {
  padding: 1em;
}
