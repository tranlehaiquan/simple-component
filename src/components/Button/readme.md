# Button component

asds

~~~js
<Button>abc</Button>
~~~
