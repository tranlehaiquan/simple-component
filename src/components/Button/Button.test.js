import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

test('renderButton', () => {
	const component = renderer.create(
		<Button>ABC</Button>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
