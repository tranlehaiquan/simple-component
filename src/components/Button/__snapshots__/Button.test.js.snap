// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderButton 1`] = `
<button
  autoFocus={false}
  className="sp-button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  ABC
</button>
`;
